docs/sdk/v3/guides/liquidity/04-swap-and-add-liquidity.md

```diff
@@ -0,0 +1,97 @@
+```
+
+Thirdly, we construct two instances of `CurrencyAmount`s, each of which represents the initial balance of the token that we wish to swap-and-add, where the token is the token in our target liquidity pool. We use the configuration parameters of the guide to set those:
+
+```js reference title="Constructing the two CurrencyAmounts" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L60-L74
+```
+
+
+Finally we construct the position, a position object that contains the details of the position for which to add liquidity. The position liquidity can be set to 1, since liquidity is still unknown and will be set inside the call to `routeToRatio`:
+
+```js reference title="Making the call to routeToRatio" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L76-L79
+```
+
+
+### Making the swap-and function call
+
+Having constructed all the parameters we need to call `routeToRatio`, we can now make the call to the function:
+
+```js reference title="Constructing the position object" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L81-L87
+```
+The return type of the function call is [SwapToRatioResponse](https://github.com/Uniswap/smart-order-router/blob/97c1bb7cb64b22ebf3509acda8de60c0445cf250/src/routers/router.ts#L121). If a route was found successfully, this object will have two fields: the status will be set to success and the result will contain the `SwapToRatioRoute` object. We check to make sure that both of those conditions hold true before we construct and submit the transaction:
+
+```js reference title="Checking that a route was found" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L89-L94
+```
+
+### Constructing and executing the transaction to swap-and-add liquidity
+
+After making sure that a route was successfully found, we can now construct and send the transaction. The response result, that is of type`SwapToRatioRoute`, has the properties we need to construct our transaction object:
+
+```js reference title="Constructing and sending the transaction" referenceLinkText="View on Github" customStyling
+https://github.com/Uniswap/examples/blob/a34ecd48c95c075bfbc443af4b4150b481e87b8b/v3-sdk/swap-and-add-liquidity/src/example/Example.tsx#L96-L104
+```
+
+If the transaction was successful, the effect of it is to swap-and-add the the amount of each token that we supplied.
```

Comment:
```suggestion
If the transaction was successful, our swap-and-add will be completed! We should see our input token balances decrease and our position balance should be increased accordingly.
```

Adding this, we should add position balance as well as fees when we do that step of code polish